### PR TITLE
Support exponential backoff for istio-agent fetch wasm

### PR DESCRIPTION
**Pull request by wzshiming**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pkg/wasm/httpfetcher.go [hunk 1]
```diff
@@ -51,14 +51,16 @@ func (f *HTTPFetcher) Fetch(url string, timeout time.Duration) ([]byte, error) {
 		}
```
Comment (mandarjog) [inline pkg/wasm/httpfetcher.go:51]: Also remove  the Todo in line 32

File: pkg/wasm/httpfetcher.go [hunk 2]
```diff
@@ -51,14 +51,16 @@ func (f *HTTPFetcher) Fetch(url string, timeout time.Duration) ([]byte, error) {
...
 	var lastError error
 	for attempts < 5 {
 		attempts++
 		resp, err := c.Get(url)
 		if err != nil {
 			lastError = err
 			wasmLog.Debugf("wasm module download request failed: %v", err)
-			time.Sleep(f.retryBackoff)
+			time.Sleep(retryBackoff)
+			retryBackoff <<= 1
```
Comment (mandarjog) [inline pkg/wasm/httpfetcher.go:63]: I think we have min duration / max duration in the code base somewhere, could use that.

50 ms * 32 (5 hardcoded) retries is not too bad though.

Comment (bianpengyuan) [reply pkg/wasm/httpfetcher.go:63]: Yeah let's use `github.com/cenkalti/backoff` package here, which is also used by other part of the code base.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**